test/test_config.py

```diff
@@ -3,159 +3,58 @@
-
-
-class TestGetBackendType:
-    def test_valid_backend_type(self, valid_config):
-        backend_type = valid_config.get_backend_type()
+class TestConfig:
+    def test_valid_get_config_value(self, test_config):
+        backend_type = test_config.get_config_value(APIConfigOptions.BACKEND)
         assert backend_type == "db"
 
-    def test_invalid_backend_type(self, invalid_config):
-        with pytest.raises(SystemExit):
-            invalid_config.get_backend_type()
-
-
-class TestGetDBURL:
-    def test_valid_db_url(self, valid_config):
-        db_url = valid_config.get_db_url()
-        assert db_url == "mysql+pymysql://icatdbuser:icatdbuserpw@localhost:3306/icatdb"
-
-    def test_invalid_db_url(self, invalid_config):
-        with pytest.raises(SystemExit):
-            invalid_config.get_db_url()
-
-
-class TestIsFlaskReloader:
-    def test_valid_flask_reloader(self, valid_config):
-        flask_reloader = valid_config.is_flask_reloader()
-        assert flask_reloader is False
-
-    def test_invalid_flask_reloader(self, invalid_config):
-        with pytest.raises(SystemExit):
-            invalid_config.is_flask_reloader()
-
-
-class TestICATURL:
-    def test_valid_icat_url(self, valid_config):
-        icat_url = valid_config.get_icat_url()
-        assert icat_url == "https://localhost:8181"
-
-    def test_invalid_icat_url(self, invalid_config):
-        with pytest.raises(SystemExit):
-            invalid_config.get_icat_url()
-
-
-class TestICATCheckCert:
-    def test_valid_icat_check_cert(self, valid_config):
-        icat_check_cert = valid_config.get_icat_check_cert()
-        assert icat_check_cert is False
-
-    def test_invalid_icat_check_cert(self, invalid_config):
-        with pytest.raises(SystemExit):
-            invalid_config.get_icat_check_cert()
-
-
-class TestGetLogLevel:
-    def test_valid_log_level(self, valid_config):
-        log_level = valid_config.get_log_level()
-        assert log_level == "WARN"
-
-    def test_invalid_log_level(self, invalid_config):
+    def test_invalid_get_config_value(self, test_config):
+        del test_config._config["backend"]
         with pytest.raises(SystemExit):
-            invalid_config.get_log_level()
-
-
-class TestGetLogLocation:
-    def test_valid_log_location(self, valid_config):
-        log_location = valid_config.get_log_location()
-        assert (
-            log_location == "/home/runner/work/datagateway-api/datagateway-api/logs.log"
+            test_config.get_config_value(APIConfigOptions.BACKEND)
+
+    @pytest.mark.parametrize(
+        "backend_type",
+        [
+            pytest.param("python_icat", id="Python ICAT Backend"),
+            pytest.param("db", id="Database Backend"),
+        ],
+    )
+    def test_valid_config_items_exist(self, backend_type):
+        test_config = Config(
+            path=Path(__file__).parent.parent
+            / "datagateway_api"
+            / "config.json.example",
```

Comment:
Can't we use the `test_config` fixture at the top of this file to do this?

Reply:
Good point! This test function now uses the fixture.